src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -571,3 +575,39 @@ def deploy(
             )
 
         return diff
+
+    def validate(self):
+        """Call system$validate_native_app_setup() to validate deployed Native App setup script."""
+        cc.step(f"Validating Native App setup script.")
+        stage_name = StageManager.get_standard_stage_prefix(self.stage_fqn)
+        try:
+            cursor = self._execute_query(
+                f"call system$validate_native_app_setup('{stage_name}')"
+            )
+        except ProgrammingError as err:
+            if "does not exist or not authorized" in err.msg:
```

Comment:
Since the App Pkg does not exist error is a specific error, we should possibly tighten the constraint on this condition with an `and` on the error number. 

Once case I'm thinking of but is not necessarily true today: "does not exist or not authorized" could be thrown if some setup script is missing or permissions change that setup script cannot be accessed.

Reply:
if a permission issue hides the visibility of the app package, wouldn't that be functionally equivalent to it not existing, from the PoV of the user? I can update the exception's error message to add "or is not authorized" (i'll add the errno check too)